.github/autobuild/android.sh

```diff
@@ -0,0 +1,112 @@
+
+    sudo apt-get -qq update
+    sudo apt-get -qq --no-install-recommends -y install build-essential zip unzip bzip2 p7zip-full curl chrpath \
+        libxkbcommon-x11-0 openjdk-8-jre-headless openjdk-8-jdk-headless
+}
+
+setup_android_sdk() {
+    mkdir -p "${ANDROID_BASEDIR}"
+
+    if [[ -d "${COMMANDLINETOOLS_DIR}" ]]; then
+        echo "Using commandlinetools installation from previous run (actions/cache)"
+    else
+        mkdir -p "${COMMANDLINETOOLS_DIR}"
+        curl -s -o downloadfile "https://dl.google.com/android/repository/commandlinetools-linux-${COMMANDLINETOOLS_VERSION}_latest.zip"
+        unzip -q downloadfile
+        mv cmdline-tools/* "${COMMANDLINETOOLS_DIR}"
+    fi
+
+    yes | "${ANDROID_SDKMANAGER}" --licenses
+    "${ANDROID_SDKMANAGER}" --update
+    "${ANDROID_SDKMANAGER}" "platforms;${ANDROID_PLATFORM}"
+    "${ANDROID_SDKMANAGER}" "build-tools;${ANDROID_BUILD_TOOLS}"
+}
+
+setup_android_ndk() {
+    mkdir -p "${ANDROID_BASEDIR}"
+    if [[ -d "${ANDROID_NDK_ROOT}" ]]; then
+        echo "Using NDK installation from previous run (actions/cache)"
+    else
+        curl -s -o downloadfile "https://dl.google.com/android/repository/android-ndk-${ANDROID_NDK_VERSION}-linux-x86_64.zip"
+        unzip -q downloadfile
+        mv "android-ndk-${ANDROID_NDK_VERSION}" "${ANDROID_NDK_ROOT}"
+    fi
+}
+
+setup_qt() {
+    if [[ -d "${QT_BASEDIR}" ]]; then
+        echo "Using Qt installation from previous run (actions/cache)"
+    else
+        echo "Installing Qt..."
+        python3 -m pip install "aqtinstall==${AQTINSTALL_VERSION}"
+        python3 -m aqt install-qt --outputdir "${QT_BASEDIR}" linux android "${QT_VERSION}" \
+            --archives qtbase qttools qttranslations qtandroidextras
+    fi
+}
+
+build_app_as_apk() {
+    QT_DIR="${QT_BASEDIR}/${QT_VERSION}/android"
+    MAKE="${ANDROID_NDK_ROOT}/prebuilt/${ANDROID_NDK_HOST}/bin/make"
+
+    "${QT_DIR}/bin/qmake" -spec android-clang
+    "${MAKE}" -j "$(nproc)"
+    "${MAKE}" INSTALL_ROOT=android-build -f Makefile install
+    "${QT_DIR}"/bin/androiddeployqt --input android-Jamulus-deployment-settings.json --output android-build \
+        --android-platform "${ANDROID_PLATFORM}" --jdk "${JAVA_HOME}" --gradle
+}
+
+pass_artifact_to_job() {
+    mkdir deploy
+    artifact_deploy_filename="jamulus_${jamulus_buildversionstring}_android.apk"
+    echo "Moving android-build/build/outputs/apk/debug/android-build-debug.apk to deploy/${artifact_deploy_filename}"
+    mv ./android-build/build/outputs/apk/debug/android-build-debug.apk ./deploy/"${artifact_deploy_filename}"
+    echo "::set-output name=artifact_1::${artifact_deploy_filename}"
+}
+
+case "${1:-}" in
+    setup)
+        setup_ubuntu_dependencies
+        setup_android_ndk
+        setup_android_sdk
+        setup_qt
+        ;;
+    build)
+        build_app_as_apk
+        ;;
+    get-artifacts)
+        pass_artifact_to_job
+        ;;
+    *)
+        echo "Unknown stage '${1:-}'"
```

Comment:
Although this would apply to all scripts: We should probably output something like "NOTE: This script is designed to run on GitHub Actions, not locally. Please run the deploy scripts in the respective OS folder"